packages/react-components/react-list-preview/src/components/ListItem/useListItem.tsx

```diff
@@ -0,0 +1,218 @@
+      // If it's one of the Arrows defined, jump out of the list item to focus on the ListItem itself
+      // The ArrowLeft will only trigger if the target element is the leftmost, otherwise the
+      // arrowNavigationAttributes handles it and prevents it from bubbling here.
+
+      if (e.key === ArrowLeft) {
+        dispatchGroupperMoveFocusEvent(e.target as HTMLElement, TabsterTypes.GroupperMoveFocusActions.Escape);
+      }
+
+      if (e.key === ArrowDown || e.key === ArrowUp) {
+        e.preventDefault();
+        // Press ESC on the original target to get focus to the parent group (List)
+        dispatchGroupperMoveFocusEvent(e.target as HTMLElement, TabsterTypes.GroupperMoveFocusActions.Escape);
+
+        // Now dispatch the original key to move up or down in the list
+        dispatchMoverMoveFocusEvent(e.currentTarget as HTMLElement, TabsterTypes.MoverKeys[e.key]);
+      }
+    }
+
+    // Now return early if the event is not fired from the list item itself
+    // as the following code handles the events specifically coming from the list item
+    if (e.target !== e.currentTarget) {
+      return;
+    }
+
+    // Space always toggles selection (if enabled)
+    if (e.key === Space) {
+      e.preventDefault();
+      if (isSelectionEnabled) {
+        toggleItem?.(e, value);
+      } else {
+        e.currentTarget.click();
+      }
+    }
+
+    // Handle clicking the list item when user presses the Enter key
+    // This internally triggers selection in the onClick handler if it hasn't been prevented
+    if (e.key === Enter) {
+      e.preventDefault();
+      e.currentTarget.click();
+    }
+
+    // Handle entering the list item when user presses the ArrowRight
+    if (e.key === ArrowRight) {
+      dispatchGroupperMoveFocusEvent(e.target as HTMLElement, TabsterTypes.GroupperMoveFocusActions.Enter);
+    }
+  });
+
+  const onCheckboxChange = useEventCallback((e: React.ChangeEvent<HTMLInputElement>, data: CheckboxOnChangeData) => {
+    if (!isSelectionEnabled || e.defaultPrevented) {
+      return;
+    }
+
+    toggleItem?.(e, value);
+  });
+
+  const onCheckboxClick = useEventCallback((e: React.MouseEvent<HTMLInputElement>) => {
+    e.stopPropagation();
+  });
+
+  const arrowNavigationAttributes = useArrowNavigationGroup({
+    axis: 'horizontal',
+  });
+
+  const tabsterAttributes = useMergedTabsterAttributes_unstable(
+    focusableItems ? arrowNavigationAttributes : ({} as TabsterDOMAttribute),
+    focusableGroupAttrs,
+  );
+
+  const root = slot.always(
+    getIntrinsicElementProps(DEFAULT_ROOT_EL_TYPE, {
+      ref: useMergedRefs(rootRef, ref) as React.Ref<HTMLLIElement & HTMLDivElement>,
+      tabIndex: focusableItems ? 0 : undefined,
+      role: listItemRole,
+      id: String(value),
+      ...(isSelectionEnabled && {
+        'aria-selected': isSelected,
+      }),
+      ...tabsterAttributes,
+      ...props,
+      'data-value': String(value),
```

Comment:
I don't see this used anywhere - is this cruft?

Reply:
This is used in the ListActiveElement story. While it makes it easy to work with event handlers, i think we can leave this responsibility on the user. I will remove this and update the story to pass the data-value from above.